### PR TITLE
PR for #217

### PR DESCRIPTION
**Pull request by hirako2000**
#217

### CODE REVIEW

File: examples/basic/line/script.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    position: new THREE.Vector3(0, 10, 200)
+  }),
+  controlsModule
+]);
+
+controlsModule.controls.autoRotate = true;
+
+const yPadding = 2;
+const xzPadding = 1;
+let yDist = Math.floor(Math.random() * 10) + yPadding;
```
Comment (sasha240100) [inline examples/basic/line/script.js]: New line `16-17`

File: examples/basic/line/script.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+for (let i = 1; i < 100; i++) {
+  const xStart = Math.floor(Math.random() * i * (i % 2 === 0 ? -1 : -1)) + (xzPadding * (i % 2 === 0 ? -10 : 10));
+  const xDist = Math.floor(Math.random()) + 5;
+  const zStart = Math.floor(Math.random() * i * (i % 2 === 0 ? -1 : -1)) + (xzPadding * (i % 2 === 0 ? -10 : 10));
+  yDist = Math.floor(Math.random() * 50) + (yPadding * 10);
+  const bottomPoint1 = {x: xStart, y: 0, z: zStart};
+  const bottomPoint2 = {x: xStart + xDist, y: 0, z: zStart};
+  const bottomPoint3 = {x: xStart, y: 0, z: -xDist + zStart};
+  const bottomPoint4 = {x: xStart + xDist, y: 0, z: -xDist + zStart};
```
Comment (sasha240100) [inline examples/basic/line/script.js]: I don't like such `bottomPoint1`, `bottompoint2` and so on... 

You can just make an array of `THREE.Vector3`

File: examples/basic/line/script.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      })
+    });
+
+    line.addTo(world);
+  }
+
+  // closes the top square
+  const line2 = new WHS.Line({
+    geometry: {
+      start: new THREE.Vector3(points[points.length - 1].x, points[points.length - 1].y + yDist, points[points.length - 1].z),
```
Comment (sasha240100) [inline examples/basic/line/script.js]: Too long line. Split such to:

```javascript
new THREE.Vector3(
  points[points.length - 1].x, 
  points[points.length - 1].y + yDist, 
  points[points.length - 1].z
)
```

File: examples/basic/line/script.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  const line = new WHS.Line({
+    geometry: {
+      start: new THREE.Vector3(points[points.length - 1].x, points[points.length - 1].y, points[points.length - 1].z),
+      end: new THREE.Vector3(points[0].x, points[0].y, points[0].z)
+    },
+
+    material: new THREE.LineBasicMaterial({
+      color
+    })
+  });
```
Comment (sasha240100) [inline examples/basic/line/script.js]: New line `88-89`

File: examples/basic/line/script.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+  line3.addTo(world);
+}
+
+function getRandomColor() {
+  const letters = '0123456789ABCDEF';
+  let color = '0x';
+  for (let i = 0; i < 6; i++) color += letters[Math.floor(Math.random() * 16)];
```
Comment (sasha240100) [inline examples/basic/line/script.js]: New line `124-125`

File: examples/basic/line/script.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+    material: new THREE.LineBasicMaterial({
+      color
+    })
+  });
+
+  line3.addTo(world);
+}
+
+function getRandomColor() {
```
Comment (sasha240100) [inline examples/basic/line/script.js]: `new THREE.Color(0xffffff * Math.random()).getHexString()`

https://threejs.org/docs/index.html?q=color#Reference/Math/Color.getHexString

File: examples/basic/line/script.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+  return parseInt(color, 16);
+}
+
+new WHS.Box({
+  geometry: {
+    width: 900,
+    height: 900,
+    widthSegments: 50,
+    heightSegments: 50
+  },
```
Comment (sasha240100) [inline examples/basic/line/script.js]: New line `134-135`

File: examples/basic/line/script.js [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+  position: [0, -1.5, 0],
```
Comment (sasha240100) [inline examples/basic/line/script.js]: New line `135-136`

File: examples/basic/line/script.js [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+  rotation: {
+    y: 0,
+    x: -Math.PI / 2
+  },
```
Comment (sasha240100) [inline examples/basic/line/script.js]: New line `139-140`

File: examples/basic/line/script.js [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+  shadow: {
+    cast: false,
+    receive: false
+  },
```
Comment (sasha240100) [inline examples/basic/line/script.js]: New line `143-144`

File: examples/basic/line/script.js [hunk 11]
```diff
@@ -0,0 +1,150 @@
+import * as UTILS from '../../globals';
+
+const controlsModule = new WHS.controls.OrbitModule();
+
+const world = new WHS.App([
+  ...UTILS.appModules({
+    position: new THREE.Vector3(0, 10, 200)
+  }),
```
Comment (sasha240100) [inline examples/basic/line/script.js]: New line `8-9`

File: src/components/__tests__/line.spec.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+        start: new Vector3(startVect.x, startVect.y, startVect.z),
+        end: new Vector3(endVect.x, endVect.y, endVect.z)
+      }
+    });
+
+    expect(line.geometry.vertices[0]).toEqual({
+      "x": startVect.x,
+      "y": startVect.y,
+      "z": startVect.z
+    });
```
Comment (sasha240100) [inline src/components/__tests__/line.spec.js]: New line `27-28`

File: src/components/__tests__/line.spec.js [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+    const line = new WHS.Line({
+      geometry: {
+        start: new Vector3(startVect.x, startVect.y, startVect.z),
+        end: new Vector3(endVect.x, endVect.y, endVect.z)
+      }
+    });
+
+    expect(line.geometry.vertices[0]).toEqual({
+      "x": startVect.x,
```
Comment (sasha240100) [inline src/components/__tests__/line.spec.js]: Use `x` instead of `"x"`

**Comment by sasha240100** [comment]
@hirako2000 Fix?

File: examples/basic/line/script.js (hunk 6 above)
Comment (hirako2000) [inline examples/basic/line/script.js]: nice!

**Pull request merged**